standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/utils/TestMetaStoreUtils.java

```diff
@@ -82,8 +83,9 @@ public void testDateToString() {
 
   @Test
   public void testTimestampToString() {
-    assertEquals(timestamp, MetaStoreUtils.convertTimestampToString(Timestamp.valueOf(timestamp)));
-  }
+    String expectedTimestampString = ZonedDateTime.parse(timestamp, FORMATTER.withZone(ZoneId.systemDefault())).format(FORMATTER);
+    assertEquals(expectedTimestampString, MetaStoreUtils.convertTimestampToString(Timestamp.valueOf(timestamp)));                         // In jdk17, Timestamp makes use of daylight savings which
+  }                                                                                                                                       // needs to be incorporated in expectedString too
```

Comment:
Please refer to [HIVE-28377](https://github.com/apache/hive/commit/e31811bb7c6670ab1f725adde3aa2b012ca64415) for the fix.